### PR TITLE
test(app): add scoped env harness for config tests

### PR DESCRIPTION
**Pull request by gh-xj**
## Summary

- add a shared `ScopedEnv` helper for app tests
- migrate `config/shared.rs` home-dir fallback coverage to the shared helper
- keep this PR limited to deterministic env restoration and test harness reuse

## Scope

- [x] Small and focused
- [ ] Includes docs updates (if needed)
- [x] No unrelated refactors

## Risk Track

- [x] Track A (routine/low-risk)
- [ ] Track B (higher-risk/policy-impacting)

## Validation

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo test --workspace --all-features`
- [ ] Additional scenario/benchmark checks (if applicable)
- [ ] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [x] If tests mutate process-global env: document how state is restored or serialized

Env-mutating test coverage now uses `ScopedEnv`, which takes a shared mutex, snapshots the original values once per key, and restores them on drop.

## Linked Issues

No linked issue. Split from PR #4 review follow-up.

### CODE REVIEW

File: crates/app/src/test_support.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+pub(crate) struct ScopedEnv {
+    originals: Vec<(&'static str, Option<OsString>)>,
+    _guard: MutexGuard<'static, ()>,
+}
+
+impl ScopedEnv {
+    pub(crate) fn new() -> Self {
+        let guard = env_lock().lock().expect("env lock should not be poisoned");
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/app/src/test_support.rs]: Fixed in `3f55640`: `ScopedEnv::new()` now recovers the inner guard from a poisoned mutex instead of panicking, and `test_support::tests::scoped_env_recovers_after_mutex_poison` covers the regression.
